File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        q, found_null = deque([root]), False
        while q:
            node = q.popleft()
            if not node:
                found_null = True
            else:
                if found_null == True:
                    return False
                q.append(node.left)
                q.append(node.right)
        return True

File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function(root) {
    const q = [root]
    let nullFound = false
    while (q.length) {
        const node = q.shift()
        if (!node) {
            nullFound = true
        } else {
            if (nullFound) return false
            q.push(node.left)
            q.push(node.right)
        }
    }
    return true
};